syntax = "proto3";

package fraud_detection;

import "google/protobuf/empty.proto";

service FraudDetectionService {
    rpc InitOrder (InitializationRequest) returns (google.protobuf.Empty) {}
    rpc CheckUserData (ContinuationRequest) returns (DetectionResponse) {}
    rpc CheckCreditCard (ContinuationRequest) returns (DetectionResponse) {}
}

message InitializationRequest {
    string order_id = 1;
    User user = 2;
    CreditCard credit_card = 3;
}

message ContinuationRequest {
    string order_id = 1;
}

message DetectionResponse {
    bool is_fraud = 1;
}

message User {
    string name = 1;
    string contact = 2;
    Address address = 3;
}

message Address {
    string street = 1;
    string city = 2;
    string state = 3;
    string zip = 4;
    string country = 5;
}

message CreditCard {
    string number = 1;
    string expiration_date = 2;
    string cvv = 3;
}
